packages/theme/src/cli/utilities/theme-environment/remote-theme-watcher.ts

```diff
@@ -0,0 +1,24 @@
+import {pollThemeEditorChanges} from './theme-polling.js'
+import {reconcileJsonFiles} from './theme-reconciliation.js'
+import {outputDebug} from '@shopify/cli-kit/node/output'
+import {AdminSession} from '@shopify/cli-kit/node/session'
+import {Checksum, Theme, ThemeFileSystem} from '@shopify/cli-kit/node/themes/types'
+
+export const LOCAL_STRATEGY = 'local'
+export const REMOTE_STRATEGY = 'remote'
+
+// **
+// * Synchronizes changes applied to JSON files via the theme editor to the local directory
+// * Does not upload any local changes to the remote theme
+// *
+export async function reconcileAndPollThemeEditorChanges(
+  targetTheme: Theme,
+  session: AdminSession,
+  remoteChecksums: Checksum[],
+  localThemeFileSystem: ThemeFileSystem,
+) {
+  outputDebug('Initiating theme asset reconciliation process')
+  const updatedRemoteChecksums = await reconcileJsonFiles(targetTheme, session, remoteChecksums, localThemeFileSystem)
```

Comment:
I decided to update remote checksums manually in reconcileJsonFiles

**Pros**
- Since we are not uploading any files in that step, it's a relatively straightforward update. The only operation that can modify the remote checksums is when we delete files that are only present on the remote theme
- Avoids the cost + latency of an additional API call

**Cons**
The API is the most accurate and up to date source of truth

Now that I'm thinking about it, maybe it's better to make the extra call since the DELETE operation can fail...

I'm going to leave this for now since it's an easy revert 
cc @karreiro

Reply:
The `deleteThemeAsset` function returns a boolean, so we can still reason about it and handle it in the `theme-reconciliation.ts` file

This seems to be the kind of context (since it's not recurring) where an extra API call helps keep the code more concise, so I'd follow that approach — but feel free to follow the optimization path as well; that's fair :)